spec/options/strictNumbers.spec.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+var Ajv = require('../ajv');
+
+describe('structNumbers option', function() {
+  var ajv;
+  describe('strictNumbers default', function() {
+    beforeEach(function () {
+      ajv = new Ajv();
+    });
+
+    it('should NOT fail validation for NaN/Infinity as type number', function() {
+      var validate = ajv.compile({type: 'number'});
+      validate(1).should.equal(true);
+      validate(NaN).should.equal(true);
+      validate(Infinity).should.equal(true);
+    });
+
+    it('should not impact validation for NaN/Infinity as type integer', function() {
+      var validate = ajv.compile({type: 'integer'});
+      validate(1).should.equal(true);
+      validate(NaN).should.equal(false);
+      validate(Infinity).should.equal(true);
+    });
+  });
+
+  describe('strictNumbers = false', function() {
+    beforeEach(function () {
+      ajv = new Ajv({strictNumbers: false});
+    });
+
+    it('should NOT fail validation for NaN/Infinity as type number', function() {
+      var validate = ajv.compile({type: 'number'});
+      validate(1).should.equal(true);
+      validate(NaN).should.equal(true);
+      validate(Infinity).should.equal(true);
+    });
+
+    it('should not impact validation for NaN/Infinity as type integer', function() {
+      var validate = ajv.compile({type: 'integer'});
+      validate(1).should.equal(true);
+      validate(NaN).should.equal(false);
+      validate(Infinity).should.equal(true);
+    });
+  });
+
+  describe('strictNumbers = true', function() {
+    beforeEach(function () {
+      ajv = new Ajv({strictNumbers: true});
+    });
+
+    it('should fail validation for NaN/Infinity as type number', function() {
+      var validate = ajv.compile({type: 'number'});
+      validate(1).should.equal(true);
+      validate(NaN).should.equal(false);
+      validate(Infinity).should.equal(false);
+    });
+
+    it('should not impact validation for NaN as type integer', function() {
+      var validate = ajv.compile({type: 'integer'});
+      validate(1).should.equal(true);
+      validate(NaN).should.equal(false);
+      validate(Infinity).should.equal(true);
+    });
+  });
+});
```

Comment:
It's very thorough - thanks. I suggest taking `false` and default  test into some function as it's exactly the same test for two option values - so it is clear that the behaviour for `false` and default is the same.

Something like:

```javascript
describe('strictNumbers default', testWithoutStrictNumbers(new Ajv))
describe('strictNumbers = false', testWithoutStrictNumbers(new Ajv({strictNumbers: false})))

function testWithoutStrictNumbers(_ajv) {
  return function () {
  // ...
  }
}
```